### PR TITLE
Introduce `python_version` in `tests[0].python`

### PR DESCRIPTION
**Pull request by hadim**
It's a proposal to fix https://github.com/prefix-dev/rattler-build/issues/1168

Please let me know if you would prefer another approach.

### CODE REVIEW

**Comment by hadim** [comment]
Here is the kind of recipe that now works:

```yaml
package:
  name: test-with-custom-deps
  version: 0.0.1

build:
  number: 0
  noarch: python
  script: touch $PREFIX/test-with-custom-deps

requirements:
  run:
    - python

tests:
  - python:
      imports:
        - pandas
      pip_check: true
      requirements:
        run:
          - pandas
```

**Comment by hadim** [comment]
Recipe format PR at https://github.com/prefix-dev/recipe-format/pull/43

**Comment by hadim** [comment]
@wolfv what do you think about this approach?

**Comment by wolfv** [comment]
Excellent PR, really awesome! 

Hmm I am a bit conflicted about this. The python check is pretty simplistic anyways and one can also relatively easy make better tests using the script tests. One can even write Python scripts directly in the scripts test, e.g.

```
- script:
    interpreter: python
    content:
      - import foo
      - import bar
      - print("hello world")
    requirements:
      run:
        - python
        - foo
```

Although it's clearly a bit more verbose.

Instead of the total flexibility of requirements we could also add a python version field to the python test as another idea:

```
python:
  imports:
     - foo
  python_version: 3.8  # or [3.8, 3.9, ...]
```

So in general I am a bit on the fence.

**Comment by hadim** [comment]
> The python check is pretty simplistic anyways and one can also relatively easy make better tests using the script tests.

Yes, I understand your point here.

> Instead of the total flexibility of requirements we could also add a python version field to the python test as another idea

This is an alternative solution I also proposed at https://github.com/prefix-dev/rattler-build/issues/1168. 

I think that one is better since it's specific to python and, as you said, for a more complex scenarios, the user already has other options.

---

So would you be okay if I give it a try to `python_version: 3.8  # or [3.8, 3.9, ...]`?

**Comment by hadim** [comment]
@wolfv I gave it a try to `python_version`.

No stress if you still don't want that feature. Just let me know.

File: docs/tutorials/javascript.md [hunk 1]
```diff
@@ -45,4 +45,4 @@ requirements:
 tests:
   - script:
     - bibtex-tidy --version
-```
\ No newline at end of file
+```
```
Comment (hadim) [inline docs/tutorials/javascript.md:48]: Unrelated from `pixi run lint`

**Pull request merged**